src/semantic_version.cr

```diff
@@ -47,21 +71,31 @@ class SemanticVersion
     end
   end
 
+  # :nodoc:
   def <=>(other : self) : Int32
     r = major <=> other.major
-    return r if r != 0
+    return r unless r.zero?
     r = minor <=> other.minor
-    return r if r != 0
+    return r unless r.zero?
     r = patch <=> other.patch
-    return r if r != 0
+    return r unless r.zero?
 
     pre1 = prerelease
     pre2 = other.prerelease
 
     prerelease <=> other.prerelease
   end
 
+  # Contains additional pre-release metadata related to this semantic version
   struct Prerelease
+    # Parses a `Prerelease` from the given pre-release metadata string
+    #
+    # ```
+    # require "semantic_version"
+    #
+    # prerelease = SemanticVersion::Prerelease.parse("rc.1.3")
```

Comment:
Usually you'd write `rc1` without a dot in between.

Reply:
Was going off of examples on https://semver.org/.  Seems they have it as like `2.0.0-rc.1` or `2.0.0-rc.2`

Reply:
I don't think it matters either way.